### DISCUSSION TITLE
Getting the reply from custom keyboards [Help]

### DISCUSSION THREAD

**Discussion by notverycool** [python-telegram-bot/python-telegram-bot#2598] [Q&A] [answered]
I'm trying to receive data from this custom keyboard. Data referring to the Log in and cancel. Looking through documentation it was easy setting it up but getting the data from the keyboard is proving to be too difficult to find out for me. 

```
                log_on_button = telegram.KeyboardButton(text='Log in')
		cancel_button = telegram.KeyboardButton(text='Cancel')
		custom_keyboard = [[log_on_button, cancel_button]]
		reply_markup = telegram.ReplyKeyboardMarkup(custom_keyboard)
```

Would there be a specific handler for this? Or possibly a way to access the information that was not clear in the documentation? Thanks for the help everyone ^^

**Comment by harshil21** [comment, marked as answer]
For KeyboardButtons, you can simply use a [MessageHandler](https://python-telegram-bot.readthedocs.io/en/stable/telegram.ext.messagehandler.html) to handle it since they are just regular text messages.

Have a look at [this example](https://github.com/python-telegram-bot/python-telegram-bot/blob/master/examples/conversationbot.py) to see how to work it out.

**Reply by notverycool** [reply]
I do understand the example given my usage case is too different for me to translate into my code. Sorry about that I'm really new to programming. The Code that I'm trying to create is:
```
                button_return = False
                log_on_button = telegram.KeyboardButton(text='Log on')
		cancel_button = telegram.KeyboardButton(text='Cancel')

		custom_keyboard = [[log_on_button, cancel_button]]

		reply_markup= telegram.ReplyKeyboardMarkup(custom_keyboard, one_time_keyboard=True, 
                input_field_placeholder='confirm log on?')

		context.bot.send_message(chat_id=update.effective_chat.id, text="Confirm Login?", reply_markup=reply_markup)

		MessageHandler(Filters.regex('^(Log on)$'), button_return = True)
		MessageHandler(Filters.regex('^(Canel)$'), button_return = False)

		if button_return:
                    other code here
```
sorry for the partial pseudo code I really don't know how the syntax is supposed to be placed to example this properly. Thanks for the help! ^^

**Reply by harshil21** [reply]
The MessageHandler accepts a callback parameter, which is called when the 'Log On' or 'Cancel' button is pressed. The handlers are usually defined and set at the end of the main file (in most cases anyway). Here's some code (untested):

<details>
<summary>Tiny example</summary>

```python

from telegram import *  # bad practice don't do this (used just for the purposes of this example)

def log_on(update, context):
    # Do something when user presses log on button

def cancel(update, context):
    # Do something when user presses cancel button

def send_kb(update, context):
    custom_keyboard = [['Log on', 'Cancel']]
    reply_markup= ReplyKeyboardMarkup(custom_keyboard, one_time_keyboard=True, 
                             input_field_placeholder='confirm log on?')

    update.message.reply_text(text="Confirm Login?", reply_markup=reply_markup)


updater = Updater('TOKEN')
dp = updater.dispatcher

dp.add_handler(MessageHandler(Filters.regex('^(Log on)$'), log_on)
dp.add_handler(MessageHandler(Filters.regex('^(Cancel)$'), cancel)
dp.add_handler(CommandHandler('log_kb', send_kb)

updater.start_polling()
updater.idle()
```

</details>

It seems that you're new to the library, hence its recommended to read [this wiki page](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Extensions-%E2%80%93-Your-first-Bot)

**Comment by notverycool** [comment]
Yup! This worked. Thanks for the help 😄
